mpf/devices/autofire.py

```diff
@@ -110,6 +117,17 @@ def _hit(self):
         """Rule was triggered."""
         if not self._ball_search_in_progress:
             self.config['playfield'].mark_playfield_active_from_device_action()
+        if self.config['enable_timeouts']:
+            self._timeout_hits.append(_time())
+            while True:
+                if self._timeout_hits[-1]-self._timeout_hits[0]>self._timeout_watch_time:
+                    self._timeout_hits.pop(0)
+                else:
+                    break
+            if len(self._timeout_hits)>self._timeout_max_hits:
+                self.disable()
+                _timeout_timer=_Timer(self._timeout_disable_time,self.enable)
```

Comment:
I like the approach. There is a race if the pops are disabled during the disable period. With DelayManager you could just cancel the delay in the disable function.